test/colon.jl

```diff
@@ -0,0 +1,15 @@
+using KernelAbstractions, Test
+
+@kernel function copy_colon_kernel!(A, @Const(B))
+    I = @index(Global)
+    @inbounds A[I] = sum(B[:,I])
```

Comment:
```suggestion
    @inbounds A[I] = sum(view(B, :,I))
```
maybe?

Reply:
That solves our problem in `ExaPF`, but then doesn't trigger the `unsafe_getindex` overdub in https://github.com/JuliaGPU/KernelAbstractions.jl/blob/51a59dc8ff2427b53db93a2e50efb4a972e370f5/src/compiler.jl#L94 .

Reply:
https://github.com/exanauts/ExaPF.jl/blob/34f3a7103851f04c61b578808880ad64aeba739d/src/LinearSolvers/preconditioners.jl#L277  this is another statement that breaks for the same reason but without `:`.